DevCycle/DVCClient.swift

```diff
@@ -252,8 +252,27 @@ public class DVCClient {
             }
         })
     }
-
+    
     public func variable<T>(key: String, defaultValue: T) -> DVCVariable<T> {
+        // initializing as variable with default value
+        var variable: DVCVariable<T> = DVCVariable(
+            key: key,
+            type: String(describing: T.self),
+            value: nil,
+            defaultValue: defaultValue,
+            evalReason: nil
+        )
+        if (Thread.isMainThread) {
+            variable = variableInternal(key: key, defaultValue: defaultValue)
+        } else {
+            DispatchQueue.main.sync {
+                variable = variableInternal(key: key, defaultValue: defaultValue)
+            }
+        }
+        return variable
+    }
```

Comment:
Can we simplify it to:

```suggestion
        if (Thread.isMainThread) {
            return variableInternal(key: key, defaultValue: defaultValue)
        } else {
            DispatchQueue.main.sync {
                return variableInternal(key: key, defaultValue: defaultValue)
            }
        }
    }
```

Reply:
that won't work as we have to ensure we return a value outside of the DispatchQueue
<img width="1149" alt="image" src="https://user-images.githubusercontent.com/98763537/225729942-c367975c-840d-41e1-b0fa-d4a87faebd81.png">

Reply:
hum... does this work?

```suggestion
        if (Thread.isMainThread) {
            return variableInternal(key: key, defaultValue: defaultValue)
        } else {
            let variable
            DispatchQueue.main.sync {
                variable = variableInternal(key: key, defaultValue: defaultValue)
            }
            return variable
        }
    }
```

Reply:
`variable` still needs to be initialized or else there would be nothing to return at the end of the function, so the best we could do is move the default into the else?
```
    public func variable<T>(key: String, defaultValue: T) -> DVCVariable<T> {
        if (Thread.isMainThread) {
            return variableInternal(key: key, defaultValue: defaultValue)
        } else {
            var variable = DVCVariable(
                key: key,
                type: String(describing: T.self),
                value: nil,
                defaultValue: defaultValue,
                evalReason: nil
            )
            DispatchQueue.main.sync {
                variable = variableInternal(key: key, defaultValue: defaultValue)
            }
            return variable
        }
    }
```

Reply:
<img width="1131" alt="image" src="https://user-images.githubusercontent.com/98763537/225731799-031be5e1-1469-463c-b19c-8354c3ae3d2c.png">